### DISCUSSION TITLE
Recommendation for accordions with checkboxes

### DISCUSSION THREAD

**Discussion by mindytom** [twilio-labs/paste#1153] [Q&A] [answered]
Hey DSYS!

I'm thinking of using accordions with checkbox groups in Event Streams. Are there any recommendations on how to use these components together? 

Here's a [clickable prototype](https://twilio.invisionapp.com/share/CPZXED8M5HG#/443776637_Step_4-_default) to show how I'm thinking it will work. 

I have a few questions called out in this image as well: 
![Accordion](https://user-images.githubusercontent.com/65037539/106663921-3f188b80-6559-11eb-897a-04180bbd00b1.png)

**Comment by serifluous** [comment, marked as answer]
Hey @mindytom ! This is definitely a tough problem.

Because of the 3 levels of nesting and expanding/collapsing you're trying to achieve here, what you'd need is a Tree component to appropriately communicate the hierarchy of all the children elements. Without a Tree, it'll be unclear to a screenreader whether "Workflow" is a child of "Reservation" or "Taskrouter". (We don't have a Tree yet, though, but it is on our roadmap.)

_However_ if you're able to simplify the list to 2 levels only, we'd recommend removing the column of checkboxes on the left, and putting the Checkbox Groups inside each Disclosure so a customer would expand it and see something like:

- [ ] **Select all**
    - [ ] com.twilio.voice...
    - [ ] com.twilio.voice...
    - [ ] com.twilio.voice...

This makes the relationship between the checkboxes and their labels a bit more straightforward. It could also make it a clearer what a customer is selecting at the top level (e.g., they're selecting "All Voice events", and not an item called "Voice")

You could also potentially make "Voice", "Taskrouter", and "Messaging" text [headings](https://paste.twilio.design/components/heading), and have just the children under Taskrouter show up as Disclosures.

Since the interaction you've mocked up is a little bit similar to "expandable table rows" we've seen, I'd recommend reaching out to folks like @kylegilbreath, @syeo, and @PoojMenon, who have designed alternatives to expandable rows.

Whatever your solution ends up being, please let us know where you took it! I'm sure your solution here could help out lots of folks working with complex pages like this.

**Reply by kate-porter** [reply]
Hey @mindytom - just wanted to followup and see if you had any additional questions or needed more clarity on Sarah's response? If you're all set for now, I'll go ahead and mark this one as answered. Thanks!

**Reply by mindytom** [reply]
@serifluous Thanks Sarah for the thorough response! I iterated on the design. Any further thoughts with this version? I think it makes it a lot more clear now! 

An additional question I have is around the pattern of the disclosure. Is there a recommendation on how they behave with one another? If the "Reservation" disclosure is already expanded, and the user clicks to expand the "Workflow" disclosure, is it recommended that both the disclosures remain open? Or expanding the second disclosure will also close the first disclosure? 

<img width="677" alt="Screen Shot 2021-02-11 at 2 11 59 PM" src="https://user-images.githubusercontent.com/65037539/107705129-2738ba00-6c73-11eb-9440-b63aad4fe369.png">

**Reply by corinne-nagel** [reply]
Hey @mindytom ! We had some time in our design jam sesh to [try out some different layouts](https://www.figma.com/file/SpeLpBoqN2vZv8KWcSn0J1/Feedback?node-id=121%3A3602). We haven't had a ton of opportunities to mock up realistic ideas using components in Figma so let us know what you think – don't feel like you have to use any of these (especially since we're not as involved in the UX/flow of this experience so the levels might not make sense). But let us know what you think! 🙏

**Reply by serifluous** [reply]
Re: disclosure behavior, Opening and closing disclosures have no affect on other disclosures—This is the typical behavior we've seen throughout Twilio products.

If you do need a component that only keeps one expandable section open, though, let us know! That's technically a different component (Accordion). We just haven't seen many examples of it in Twilio products, and it's a little lower down on our roadmap.

**Reply by mindytom** [reply]
Got it. Thank you DSYS team for all your help! Will follow up if I have any further questions.
